tick/base/__init__.py

```diff
@@ -10,10 +10,14 @@ def _set_mpl_backend():
     import os
     import sys
     # Set the backend to a non-interactive one for unices without X
-    if (os.name == 'posix' and 'DISPLAY' not in os.environ
-        and not (sys.platform == 'darwin'
-                 and matplotlib.get_backend() == 'MacOSX'
-                 )):
+    if (( os.name == 'posix' and 'DISPLAY' not in os.environ
+          and not
+            (sys.platform == 'darwin'
+             and matplotlib.get_backend() == 'MacOSX')
+        )
+          or ('DISPLAY' in os.environ
+             and os.environ['DISPLAY'] == '-1')
+        ):
```

Comment:
Maybe this addition should be in another PR describing clearly why it is needed?

Reply:
Oh that's what you meant by "isolate" sure that's cool